andela/phoenix-bn-backend
# Resolve configuration errors

### What does this PR do?

- set sequelizerc for transpiling to native JS
- set sequelize scripts
- set development environment executable
- seed admin user to database using environment secrete
- remove production configuration errors

### Description of Task to be completed?

- add @babel/register to sequelize configuration
- use coveralls badge in README
- add scripts for database migrations and seeders
- setup tasks running in development environment
- remove devdependencies from production executable file
- add admin status to User model
- add admin seeder

### How should this be manually tested?

- run `npm run reset-db` to drop and migrate database
- run `npm run seeders` to add admin, ensure environment secrete is set in `.env` file
- run `npm run devstart` to see app in development environment
- run `npm start` to see app in production environment

### Any background context you want to provide?

The `dev ` file is, so we can run development execute file independently of the `www` file. In the `www` file we are making use of the compiled files from our `dist` folder, while the `dev` file makes use of ES6 code from `src` folder, and we require babel dependencies at the beginning so it can compile our code. 

The reason for this is that;
-  the current `www` requires babel which is a devDependency, and app would crash in production since Heroku only install project dependencies
-  we could move the babel dependencies to production as we are doing presently, however compiling code in production is inefficient.

### What are the relevant pivotal tracker stories?
#168250646

### Implementation Plan

N/A

### Questions:

- suggest any other config errors or conventions to be resolved in your review
